kf_lib_data_ingest/common/misc.py

```diff
@@ -128,3 +130,55 @@ def multisplit(string: str, delimiters: list):
     assert_safe_type(delimiters, list)
     regexPattern = '|'.join(map(re.escape, delimiters))
     return re.split(regexPattern, string)
+
+
+def to_camel_case(snake_str):
+    """
+    Convert snake case str to camel case
+    """
+    words = snake_str.split('_')
+    return ''.join([w.title() for w in words])
+
+
+def check_service_status(url, logger=None, request_kwargs=None,
```

Comment:
Sentinel network status checking is both inefficient and unrobust. It makes additional requests that you don't use and then fails to gracefully handle later server errors.

All network requests should be automatically retried a reasonable number of times with a reasonable backoff  factor for a reasonable set of plausibly-intermittent error statuses via urllib3's Retry library with the method whitelist set to Retry.DEFAULT_METHOD_WHITELIST+{'POST', 'PATCH'}. 

See https://www.peterbe.com/plog/best-practice-with-retries-with-requests

This also should be in a different PR.

Reply:
That's fine, I can use urllib3's Retry. I think it's still relevant to this PR tho bc I need it to get the schemas used in `handle_nulls`.